worldengine/draw.py

```diff
@@ -277,74 +275,42 @@ def get_biome_color_based_on_elevation(world, elev, x, y):
         ## Generate some random noise to apply to this pixel
         #  There is noise for each element of the rgb value
         #  This noise will be further modified by the height of this tile
-        noise = (numpy.random.randint(-NOISE_RANGE, NOISE_RANGE), 
-                 numpy.random.randint(-NOISE_RANGE, NOISE_RANGE), 
-                 numpy.random.randint(-NOISE_RANGE, NOISE_RANGE))
-
-        ####### Case 1 - elevation is very high ########
-        if elev > HIGH_MOUNTAIN_ELEV:     
-            # Modify the noise to make the area slightly brighter to simulate snow-topped mountains.
-            noise = add_colors(noise, HIGH_MOUNTAIN_NOISE_MODIFIER)
-            # Average the biome's color with the MOUNTAIN_COLOR to tint the terrain
-            biome_color = average_colors(biome_color, MOUNTAIN_COLOR)
-
-        ####### Case 2 - elevation is high ########
-        elif elev > MOUNTAIN_ELEV:   
-            # Modify the noise to make this tile slightly darker, especially draining the green
-            noise = add_colors(noise, MOUNTAIN_NOISE_MODIFIER)
-            # Average the biome's color with the MOUNTAIN_COLOR to tint the terrain
-            biome_color = average_colors(biome_color, MOUNTAIN_COLOR)
-
-        ####### Case 3 - elevation is somewhat high ########
-        elif elev > HIGH_HILL_ELEV:   
-            noise = add_colors(noise, HIGH_HILL_NOISE_MODIFIER)
-
-        ####### Case 4 - elevation is a little bit high ########
-        elif elev > HILL_ELEV:   
-            noise = add_colors(noise, HILL_NOISE_MODIFIER)
+        noise = rng.randint(-NOISE_RANGE, NOISE_RANGE, size=3)  # draw three random numbers at once
+
+    ####### Case 1 - elevation is very high ########
+    if world.is_high_mountain((x, y)):
+        # Modify the noise to make the area slightly brighter to simulate snow-topped mountains.
+        noise = add_colors(noise, HIGH_MOUNTAIN_NOISE_MODIFIER)
+        # Average the biome's color with the MOUNTAIN_COLOR to tint the terrain
+        biome_color = average_colors(biome_color, MOUNTAIN_COLOR)
+
+    ####### Case 2 - elevation is high ########
+    elif world.is_medium_mountain((x, y)):
+        # Modify the noise to make this tile slightly darker, especially draining the green
+        noise = add_colors(noise, MOUNTAIN_NOISE_MODIFIER)
+        # Average the biome's color with the MOUNTAIN_COLOR to tint the terrain
+        biome_color = average_colors(biome_color, MOUNTAIN_COLOR)
+
+    ####### Case 3 - elevation is somewhat high ########
+    elif world.is_low_mountain((x, y)):
+        noise = add_colors(noise, HIGH_HILL_NOISE_MODIFIER)
+
+    ####### Case 4 - elevation is a little bit high ########
+    elif world.is_hill((x, y)):
+        noise = add_colors(noise, HILL_NOISE_MODIFIER)
```

Comment:
All the static height-checks have been replaced by calls to the appropriate functions; output currently looks different than before but I suspect that is due to the parameters guessed in ```generation.py```.

Here the output of a test-world:
![satellite_28070](https://cloud.githubusercontent.com/assets/12257447/11033883/4487da72-86e7-11e5-97c1-bbcdb2c95958.png)